File: codes/2024/36-Excellence-SZU-91-0-25 (C++)/src/frontend/Token.cpp
#include "../frontend/Token.h"
#include "../utility/Diagnostor.h"

DataType Token::ToDataType()
{
    DataType data_type;

    switch(token_type)
    {
        case T_VOID:  data_type=VOID;break;
        case T_INT:   data_type=I32; break;
        case T_FLOAT: data_type=F32; break;
        default:CompilerError("Not exist corresponding data type");
    }

    return data_type;
}

File: codes/2023/4-Second-Beihang (Java)/src/Frontend/Token.java
package Frontend;

public class Token {
    private TokenType type;
    private String val;

    public Token(TokenType type, String val){
        this.type = type;
        this.val = val;
    }

    public TokenType getType() {
        return type;
    }

    public String getVal() {
        return val;
    }
}
